py/visdom/server.py

```diff
@@ -97,13 +99,15 @@ def serialize_all(state, env_path=DEFAULT_ENV_PATH):
 
 
 class Application(tornado.web.Application):
-    def __init__(self, port=DEFAULT_PORT, env_path=DEFAULT_ENV_PATH, readonly=False):
+    def __init__(self, port=DEFAULT_PORT, env_path=DEFAULT_ENV_PATH, readonly=False, with_login=False, user=False):
```

Comment:
Rather than require passing a with_login parameter, you can use `user=None` and then check in the init method if `user` is set, and use that to determine if `with_login` is true. 

Also please use `login_enabled` and `user_credentials` as your variable names.